
diff --git a/src/features/fwht/fwht_norm_avx2.hpp b/src/features/fwht/fwht_norm_avx2.hpp
index ed88088..0960555 100644
--- a/src/features/fwht/fwht_norm_avx2.hpp
+++ b/src/features/fwht/fwht_norm_avx2.hpp
@@ -124,7 +124,7 @@
inline void fwht16_norm_flat_avx2(float x[], float y[], const __m256 factor) {
//
//
//
-inline void fwht16_norm_term_avx2(__m256 X0, __m256 X1, __m256 X2, __m256 X3, float y[], const __m256 factor) {
+inline void fwht32_norm_term_avx2(__m256 X0, __m256 X1, __m256 X2, __m256 X3, float y[], const __m256 factor) {
     const __m256 A0 = X0 + X2, A1 = X1 + X3;
     const __m256 B0 = X0 - X2, B1 = X1 - X3;
     fwht16_norm_term_avx2(A0, A1, y + 0, factor);
@@ -162,8 +162,8 @@ inline void fwht64_norm_terminale_avx2(__m256 X0, __m256 X1, __m256 X2, __m256 X
     const __m256 B0 = X0 - X4, B1 = X1 - X5;
     const __m256 B2 = X2 - X6, B3 = X3 - X7;

-    fwht16_norm_term_avx2(A0, A1, A2, A3, y + 0, factor);
-    fwht16_norm_term_avx2(B0, B1, B2, B3, y + 32, factor);
+    fwht32_norm_term_avx2(A0, A1, A2, A3, y + 0, factor);
+    fwht32_norm_term_avx2(B0, B1, B2, B3, y + 32, factor);
}
//
//
@@ -187,8 +187,8 @@
inline void fwht64_norm_flat_avx2(float x[], float y[], const __m256 factor) {
     const __m256 B0 = X0 - X4, B1 = X1 - X5;
     const __m256 B2 = X2 - X6, B3 = X3 - X7;

-    fwht16_norm_term_avx2(A0, A1, A2, A3, y + 0, factor);
-    fwht16_norm_term_avx2(B0, B1, B2, B3, y + 32, factor);
+    fwht32_norm_term_avx2(A0, A1, A2, A3, y + 0, factor);
+    fwht32_norm_term_avx2(B0, B1, B2, B3, y + 32, factor);
}
//
//
@@ -497,4 +497,4 @@ template <> inline void fwht_norm_avx2<1024>(float x[]) {
/////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//
-#endif
\ No newline at end of file
+#endif
 